app/javascript/app/phone-validation.js

```diff
@@ -0,0 +1,37 @@
+import { isValidNumber } from 'libphonenumber-js';
+
+const checkPhoneValidity = () => {
+  const sendCodeButton = document.querySelector('[data-international-phone-form] input[name=commit]');
+  const phoneInput = document.querySelector('[data-international-phone-form] .phone');
+  const countryCodeInput = document.querySelector('[data-international-phone-form] .international-code');
+
+  if (phoneInput && countryCodeInput && sendCodeButton) {
+    const phone = phoneInput.value;
+    const countryCode = countryCodeInput.value;
+
+    const phoneValid = isValidNumber(phone, countryCode);
+
+    if (phoneValid) {
+      phoneInput.classList.add('valid-phone');
```

Comment:
What is the purpose of adding / removing these classes here?

Reply:
For this PR, this is a pre-optimization. I was planning to add CSS styles to add an indicator following the input field using those classes to manage it, but in review, it looked like there was enough useful in this PR as it is without waiting on that. I'll remove the classes for this PR and put them back in when I get the CSS ironed out.

Reply:
Cool, my next question was about how the user is supposed to determine why the "Send Code" button is disabled, but I suppose that is coming later.

Reply:
-nod- -- I'm not sure I want to show an error message when they're still entering their phone number, but we could do something on-blur if the number doesn't look valid. That might be a better next step before I get the CSS whipped up since we already have a mechanism for doing messaging like that.